app/models/sent_scheduled_message.rb

```diff
@@ -9,4 +9,20 @@ class SentScheduledMessage < ActiveRecord::Base
   validates :sent_on, presence: true
 
   delegate :slack_username, to: :employee
+
+  def self.filter(params)
+  	@employee = Employee.find_by slack_username: params[:slack_username]
+
+    results = self.all.where("lower(message_body) like ?", "%#{params[:message_body].downcase}%")
+
+    if @employee
```

Comment:
I'm trying this out, and I seem to need the explicit return. I arrived at this _slightly_ better syntax:
```
    @employee = Employee.find_by slack_username: params[:slack_username]

    results = self.all.where("lower(message_body) like ?", "%#{params[:message_body].downcase}%")

    if @employee
      employee_id = @employee.id
      results.where(employee_id: employee_id)
    end

    if !params[:sent_on].blank?
      results.where(sent_on: params[:sent_on])
    end

    return results
```